lib/opentelemetry_tesla.ex

```diff
@@ -61,65 +61,71 @@ defmodule OpentelemetryTesla do
     )
   end
 
-  defp handle_start(_event, _measurements, _metadata, _config) do
+  defp handle_start(_event, _measurements, %{env: %Tesla.Env{method: method}} = metadata, _config) do
+    http_method = http_method(method)
+
     OpentelemetryTelemetry.start_telemetry_span(
       @tracer_id,
-      "External HTTP Request",
-      %{},
-      %{kind: :server}
+      "HTTP #{http_method}",
+      metadata,
+      %{kind: :client}
     )
   end
 
-  defp handle_stop(_event, %{duration: measurement}, metadata, _config) do
-    span_args =
-      metadata
-      |> headers_span_args()
-      |> :lists.append(span_args(metadata))
-      |> :lists.append([{"http.request.measurement", measurement}])
-
-    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, %{})
+  defp handle_stop(_event, _measurements, metadata, _config) do
+    span_attrs = build_attrs(metadata)
+    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, metadata)
 
-    OpenTelemetry.Span.set_attributes(ctx, span_args)
+    OpenTelemetry.Span.set_attributes(ctx, span_attrs)
 
-    OpentelemetryTelemetry.end_telemetry_span(@tracer_id, %{})
+    OpentelemetryTelemetry.end_telemetry_span(@tracer_id, metadata)
   end
 
   defp handle_exception(
          _event,
          %{duration: native_time},
-         %{kind: kind, reason: reason, stacktrace: stacktrace},
+         %{kind: kind, reason: reason, stacktrace: stacktrace} = metadata,
          _config
        ) do
-    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, %{})
+    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, metadata)
 
     exception = Exception.normalize(kind, reason, stacktrace)
 
     Span.record_exception(ctx, exception, stacktrace, duration: native_time)
```

Comment:
What's the `duration` here used for? Are you wanting to override it on purpose?

Reply:
nevermind, gonna remove it.